core/src/main/kotlin/com/tschuchort/compiletesting/EncodingUtils.kt

```diff
@@ -0,0 +1,79 @@
+package com.tschuchort.compiletesting
+
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+
+/**
+ * Encode the given URI [path] with UTF_8 encoding
+ */
+fun encodePath(path: String): String {
```

Comment:
This is a lot of complicated code. Is there not a class in the standard library that can do it?

Reply:
Actually, Java has [`URLEncoder`](https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html), but it encodes string as per the `application/x-www-form-urlencoded` MIME format which is not suitable in our case.

I've taken the reference of this complex code from the [Spring framework](https://github.com/spring-projects/spring-framework/blob/main/spring-web/src/main/java/org/springframework/web/util/UriUtils.java#L179).